test/perf/benchmarks/README.md

```diff
@@ -0,0 +1,52 @@
+## Micro Benchmarks
+
+This folder contains micro benchmarks that test the performance of PowerShell Engine.
+
+### Requirement
+
+1. A good suite of benchmarks
+   Something that measures only the thing that we are interested in and _produces accurate, stable and repeatable results_.
+2. A set of machine with the same configurations.
+3. Automation for regression detection.
+
+### Design Decision
+
+1. This project is internal visible to `System.Management.Automation`.
+   We want to be able to target some internal APIs to get measurements on specific scoped scenarios,
+   such as measuring the time to compile AST to a delegate by the compiler.
+2. This project makes `ProjectReference` to other PowerShell assemblies.
+   This makes it easy to run benchmarks with the changes made in the codebase.
+   To run benchmarks with a specific version of PowerShell,
+   just replace the `ProjectReference` with a `PackageReference` to the `Microsoft.PowerShell.SDK` NuGet package of the corresponding version.
+
+### Quick Start
+
+To run the benchmarks in Interactive Mode, where you will be asked which benchmark(s) to run:
+```
+dotnet run -c release
+```
+
+To list all available benchmarks ([read more](https://github.com/dotnet/performance/blob/main/docs/benchmarkdotnet.md#Listing-the-Benchmarks)):
+```
+dotnet run -c release --list [flat/tree]
+```
+
+To filter the benchmarks using a glob pattern applied to `namespace.typeName.methodName` ([read more](https://github.com/dotnet/performance/blob/main/docs/benchmarkdotnet.md#Filtering-the-Benchmarks)]):
+```
+dotnet run -c Release -f net6.0 --filter *parser* --list flat
+```
+
+To profile the benchmarked code and produce an ETW Trace file ([read more](https://github.com/dotnet/performance/blob/main/docs/benchmarkdotnet.md#Profiling))
+```
+dotnet run -c Release -f net6.0 --filter *parser* --profiler ETW
+```
+
+### Regression Detection
+
+We use the tool [`ResultsComparer`](../dotnet-tools/ResultsComparer) to compare the provided benchmark results.
+See the [README.md](../dotnet-tools/ResultsComparer/README.md) for `ResultsComparer` for more details.
+
+## References
+
+- [Getting started with BenchmarkDotnet](https://benchmarkdotnet.org/articles/guides/getting-started.html)
```

Comment:
Please consider adding https://github.com/dotnet/performance/blob/main/docs/microbenchmark-design-guidelines.md to the list

```suggestion
- [Getting started with BenchmarkDotnet](https://benchmarkdotnet.org/articles/guides/getting-started.html)
- [Microbenchmark Design Guidelines](https://github.com/dotnet/performance/blob/main/docs/microbenchmark-design-guidelines.md)
```